content/blog/2018/06/2018-06-17-running-jenkins-with-java10-11.adoc

```diff
@@ -0,0 +1,125 @@
+
+```shell
+${JAVA10_HOME}/bin/java --add-modules java.xml.bind -jar jenkins.war \
+    --enable-future-java --httpPort=8080 --prefix=/jenkins
+```
+
+==== Java 11
+
+1. Download Jenkins WAR for 2.127 or above
+** You can also use experimental branches referenced above
+2. Download the following libraries to the same directory as jenkins.war
+** link:http://central.maven.org/maven2/javax/xml/bind/jaxb-api/2.3.0/jaxb-api-2.3.0.jar[jaxb-api-2.3.0.jar] (save as jaxb-api.jar)
+** link:http://central.maven.org/maven2/com/sun/xml/bind/jaxb-core/2.3.0.1/jaxb-core-2.3.0.1.jar[jaxb-core-2.3.0.1.jar] (save as jaxb-core.jar)
+** link:http://central.maven.org/maven2/com/sun/xml/bind/jaxb-impl/2.3.0.1/jaxb-impl-2.3.0.1.jar[jaxb-impl-2.3.0.1.jar] (save as jaxb-impl.jar)
+** https://github.com/javaee/activation/releases/download/JAF-1_2_0/javax.activation.jar[javax.activation v.1.2.0]  (save as javax.activation.jar)
+3. Run the following command:
+
+```shell
+Run Jenkins with ${JAVA11_HOME}/bin/java \
+    -p jaxb-api.jar:javax.activation.jar --add-modules java.xml.bind,java.activation \
+    -cp jaxb-core.jar:jaxb-impl.jar \
+    -jar jenkins.war --enable-future-java --httpPort=8080 --prefix=/jenkins
+```
+
+=== Current state
+
+As of June 17, we have achieved the following state:
+
+* Jenkins starts up successfully with Java 10 and 11
+* It is possible to configure and run freestyle jobs
+* Jenkins agents are able to start on Java 10, to connect to the master and to execute Freestyle jobs
+* Jenkins is able to execute basic Groovy scripts in Script Console and Groovy hooks
+* Pipeline crashes immediately on Java 10 and 11 (link:https://issues.jenkins-ci.org/browse/JENKINS-46602[JENKINS-46602])
+* JobDSL works well on demo projects
+* Git Client plugin 2.7.2 cannot be installed with Java 11,
+`3.0.0-beta3` from link:/doc/developer/publishing/releasing-experimental-updates/#configuring-jenkins-to-use-experimental-update-center[Experimental Update Center] is required
+(link:https://issues.jenkins-ci.org/browse/JENKINS-51871[JENKINS-51871])
+* There are many warnings about Illegal reflective access during execution
+(linked in link:https://issues.jenkins-ci.org/browse/JENKINS-40689[JENKINS-40689])
+
+We anticipate to discover and report more issues during the hackathon this week.
+
+=== Contributing
+
+If you discover incompatibilities in plugins, please
+link:https://wiki.jenkins.io/display/JENKINS/How+to+report+an+issue[report issues in our bugtracker].
+We have `java10` and `java11` labels for such issues.
+
+If you are interested to try out Jenkins with Java 10 and 11 before June 22nd,
+you may be interested to sign-up to the link:/blog/2018/06/08/jenkins-java10-hackathon/[Jenkins and Java 10+ online hackathon].
+Everybody is welcome to join, independently of their Jenkins experience and amount of time they have available.
+Exploratory testing is also within the hackathon's scope.
+During this event, please also use the `java10_hackathon` label.
```

Comment:
Unexpected underscore in label, have we ever used that?

Reply:
Yes, some labels in JIRA have underscores.